src/muxer/muxer_pass.c

```diff
@@ -446,10 +447,34 @@ pass_muxer_reconfigure(muxer_t* m, const struct streaming_start *ss)
     pm->pm_dst_onid  = pm->pm_src_onid;
   }
   pm->pm_pmt_pid     = ss->ss_pmt_pid;
+  pm->pm_rewrite_pmt = !!pm->m_config.u.pass.m_rewrite_pmt;
   pm->pm_rewrite_sdt = !!pm->m_config.u.pass.m_rewrite_sdt;
   pm->pm_rewrite_nit = !!pm->m_config.u.pass.m_rewrite_nit;
   pm->pm_rewrite_eit = !!pm->m_config.u.pass.m_rewrite_eit;
 
+  if (pm->pm_pmt_pid == DVB_SDT_PID && pm->pm_rewrite_pmt) {
+    tvhwarn(LS_PASS, "PMT PID shared with A/V, rewrite disabled");
+    pm->pm_rewrite_pmt = 0;
```

Comment:
I think that you should disable SDT rewrite here? PMT rewrite should be ok.

Reply:
No, since SDT and PMT share the same PID, any one of the two rewrites will destroy the stream. I always have to disable both rewrites to make the stream work.

Reply:
The `dvb_table_parse(&pm->pm_pmt, "-", tsb, l, 1, 0, pass_muxer_pmt_cb);` call should pick just PMT TS packets and filter others. It would be nice to analyze the PMT output when PMT rewrite is active, but SDT rewrite is off.

Also, it may be possible to enable both PMT and SDT rewrite with this code change:

<pre>
--- a/src/muxer/muxer_pass.c
+++ b/src/muxer/muxer_pass.c
@@ -652,20 +652,27 @@ pass_muxer_write_ts(muxer_t *m, pktbuf_t *pb)
         } else if (pid == DVB_SDT_PID) {
         
           dvb_table_parse(&pm->pm_sdt, "-", tsb, l, 1, 0, pass_muxer_sdt_cb);
+          if (pid == pm->pm_pmt_pid)
+            goto __pmt;
 
         /* NIT */
         } else if (pid == DVB_NIT_PID) {
         
           dvb_table_parse(&pm->pm_nit, "-", tsb, l, 1, 0, pass_muxer_nit_cb);
+          if (pid == pm->pm_pmt_pid)
+            goto __pmt;
 
         /* EIT */
         } else if (pid == DVB_EIT_PID) {
         
           dvb_table_parse(&pm->pm_eit, "-", tsb, l, 1, 0, pass_muxer_eit_cb);
+          if (pid == pm->pm_pmt_pid)
+            goto __pmt;
 
         /* PMT */
         } else {
 
+__pmt:
           dvb_table_parse(&pm->pm_pmt, "-", tsb, l, 1, 0, pass_muxer_pmt_cb);
 
         }
</pre>

Reply:
I tried your suggestion and it worked as well. I have to little experience to decide which one should be implemented. If you prefer this way, I can modify my pull request.

Reply:
What @perexg suggested there is actually allowing it to work when the PIDs are the same. I think we could get rid of the goto aswell by simply replacing the `else` with a `if(pid == pm->pm_pmt_pid)`.

Reply:
You are right, when replacing the `else` with an `if`, you get the same results as with the goto. In this case, the complete change starts making much more sense: PAT, SDT, NIT and EIT all have different PIDs and will never get in conflict with each other. PMT is different, though. PMT can have every PID (except 0x00, of course). Thus, it should not be treated in an `else` but should get it's separate `if`.